### PR TITLE
Document show_program and show_vcc functions [DOC-144]

### PR DESCRIPTION
**Pull request by romainbrenguier**
This also add static where missing in the corresponding cpp file.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [na] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [na] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [na] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [na] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Comment by owen-mc-diffblue** [comment]
@romainbrenguier clang-format is unhappy

File: src/goto-symex/show_vcc.cpp [hunk 1]
```diff
@@ -23,6 +23,10 @@ Author: Daniel Kroening, kroening@kroening.com
 #include <util/json_irep.h>
 #include <util/ui_message.h>
 
+/// Output equations from \p equation in plain text format to the given output
+/// stream \p out.
+/// The format is a counter value (enclosed in `{-xxx}`), followed by the
```
Comment (tautschnig) [inline src/goto-symex/show_vcc.cpp]: I struggled to understand the "enclosed in" part - how about "Each equation is prefixed by a negative index (formatted `{-N}`)."

File: src/goto-symex/show_vcc.h [hunk 2]
```diff
@@ -17,6 +17,12 @@ Author: Daniel Kroening, kroening@kroening.com
 class optionst;
 class symex_target_equationt;
 
+/// Output equations from \p equation to a file or to the standard output.
+/// The name of the output file is given by the `outfile` option from
+/// \p options, the standard input is used if it is not provided.
+/// The format is either JSON or plain text depending on \c ui_message_handler;
```
Comment (tautschnig) [inline src/goto-symex/show_vcc.h]: I'd use `\p` instead of `\c`

**Review by tautschnig — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: 08f770a4b41a70ad18108f4a0c498582fe0c6e22).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/99764909
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: b2c938ca722459b29dca8dd17261dff5e80fbda7).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/99766947

**Pull request merged**